test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreBuildTargetsAndPropsTests.cs

```diff
@@ -86,6 +87,95 @@ public async Task RestoreBuildTargetsAndProps_VerifyPropsAndTargetsGeneratedAsyn
             }
         }
 
+        [Fact]
+        public async Task RestoreBuildTargetsAndProps_VerifyPropsAndTargetsGeneratedAsyncWithFilesModifiedAfterRestore_Success()
+        {
+            // Arrange
+            using (var cacheContext = new SourceCacheContext())
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var logger = new TestLogger();
+                var sources = new List<PackageSource>
+                {
+                    new PackageSource(pathContext.PackageSource)
+                };
+
+                var spec = GetProject("projectA", "net462", "netstandard1.6");
+
+                spec.RestoreMetadata.CrossTargeting = true;
+                spec.Dependencies.Add(new LibraryDependency()
+                {
+                    LibraryRange = new LibraryRange("x", VersionRange.Parse("1.0.0"), LibraryDependencyTarget.Package)
+                });
+
+                // Create fake projects, the real data is in the specs
+                var projects = CreateProjectsFromSpecs(pathContext, spec);
+
+                // Create dg file
+                var dgFile = new DependencyGraphSpec();
+                dgFile.AddProject(spec);
+                dgFile.AddRestore(spec.RestoreMetadata.ProjectUniqueName);
+                dgFile.Save(Path.Combine(pathContext.WorkingDirectory, "out.dg"));
+
+                var packageX = new SimpleTestPackageContext()
+                {
+                    Id = "x",
+                    Version = "1.0.0"
+                };
+
+                packageX.AddFile("build/x.targets");
+                packageX.AddFile("build/x.props");
+                packageX.AddFile("contentFiles/any/any/_._");
+
+                await SimpleTestPackageUtility.CreatePackagesAsync(pathContext.PackageSource, packageX);
+
+                var project = projects[0];
+                var importGroup = "ImportGroup";
+                var condition = "Condition";
+                var internalSubset = @"<!ENTITY greeting ""Hello"">
+   <!ENTITY name ""NuGet Client "">
+   <!ENTITY sayhello ""&greeting; &name;"">";
+                var newValue = "&sayhello;";
+
+                var summaries = await RunRestoreAsync(pathContext, logger, sources, dgFile, cacheContext);
+
+                var targetsXML = XDocument.Parse(File.ReadAllText(project.TargetsOutput));
```

Comment:
Can you explain in more detail what you are trying to test here?

Reply:
> Can you explain in more detail what you are trying to test here?

During restore, `props` and `targets` files were created only if those files doesn't exist or the XML content in the files have changed [source code link] https://github.com/NuGet/NuGet.Client/blob/59a06a25c9c30f484be2357398166af0dbfd25a3/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs#L334 This test ensures that, executing restore for the second time detects that `props` and `targets` files were modified and as a result the files are overwritten.

Reply:
I think this test does not really test what you intend it to.
If you look at the implementation of RunRestoreAsync, it runs with restore no-op disabled. 

So a complete restore happens and ends up here: https://github.com/NuGet/NuGet.Client/blob/715edd1291b089d42cfef65566146f45a8cd6310/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs#L175
and later
https://github.com/NuGet/NuGet.Client/blob/715edd1291b089d42cfef65566146f45a8cd6310/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs#L324

That method itself doesn't have unit test but has some integration tests so it's already covered. 

In reality no-op itself does not check the content of the msbuild targets/props on purpose, rather it just checks the existence of the files. https://github.com/NuGet/NuGet.Client/blob/715edd1291b089d42cfef65566146f45a8cd6310/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs#L134-L145.